src/compressed-textures/dds/parseDDS.ts

```diff
@@ -0,0 +1,175 @@
+import type { TEXTURE_FORMATS } from '../../rendering/renderers/shared/texture/const';
+import type { TextureSourceOptions } from '../../rendering/renderers/shared/texture/sources/TextureSource';
+
+// All values and structures referenced from:
+// http://msdn.microsoft.com/en-us/library/bb943991.aspx/
+const DDS_MAGIC = 0x20534444;
+const DDSD_MIPMAPCOUNT = 0x20000;
+const DDPF_FOURCC = 0x4;
+
+/**
+ * @param value
+ */
+function fourCCToInt32(value: string)
+{
+    return value.charCodeAt(0)
+        + (value.charCodeAt(1) << 8)
+        + (value.charCodeAt(2) << 16)
+        + (value.charCodeAt(3) << 24);
+}
+
+/**
+ * @param value
+ */
+function int32ToFourCC(value: number)
+{
+    return String.fromCharCode(
+        value & 0xff,
+        (value >> 8) & 0xff,
+        (value >> 16) & 0xff,
+        (value >> 24) & 0xff,
+    );
+}
+
+const FOURCC_DXT1 = fourCCToInt32('DXT1');
+const FOURCC_DXT3 = fourCCToInt32('DXT3');
+const FOURCC_DXT5 = fourCCToInt32('DXT5');
+
+const headerLengthInt = 31; // The header length in 32 bit ints
+
+// Offsets into the header array
+const offMagic = 0;
+
+const offSize = 1;
+const offFlags = 2;
+const offHeight = 3;
+const offWidth = 4;
+
+const offMipmapCount = 7;
+
+const offPfFlags = 20;
+const offPfFourCC = 21;
+const offRGBBitCount = 22;
+const offRBitMask = 23;
+const offGBitMask = 24;
+const offBBitMask = 25;
+
+export function parseDDS(buffer: ArrayBuffer, supportedFormats: TEXTURE_FORMATS[]): TextureSourceOptions<Uint8Array[]>
+{
+    const header = new Int32Array(buffer, 0, headerLengthInt);
```

Comment:
Just wondering. If this function throws, does the caller have a strategy? How would a user safeguard loading faulty textures at runtime? Can they catch this with `Asset.load`?

Reply:
yoo! yes, this error should get caught by the Loader